packages/components/src/components/domain/seo/meta/__tests__/RouteMeta.test.tsx

```diff
@@ -0,0 +1,183 @@
+
+      const alternateEn = document.head.querySelector(
+        'link[rel="alternate"][hrefLang="en"]'
+      );
+      expect(alternateEn?.getAttribute('href')).toBe(`${origin}/en/test-path`);
+
+      const alternateXDefault = document.head.querySelector(
+        'link[rel="alternate"][hrefLang="x-default"]'
+      );
+      expect(alternateXDefault?.getAttribute('href')).toBe(
+        `${origin}/fi/test-path`
+      );
+
+      const ogLocale = document.head.querySelector(
+        'meta[property="og:locale"]'
+      );
+      expect(ogLocale?.getAttribute('content')).toBe('fi');
+
+      const ogUrl = document.head.querySelector('meta[property="og:url"]');
+      expect(ogUrl?.getAttribute('content')).toBe(`${origin}/fi/test-path`);
+
+      const twitterUrl = document.head.querySelector(
+        'meta[name="twitter:url"]'
+      );
+      expect(twitterUrl?.getAttribute('content')).toBe(
+        `${origin}/fi/test-path`
+      );
+    });
+
+    it('should handle root path correctly', () => {
+      router.locale = 'en';
+      router.asPath = '/';
+
+      render(<RouteMeta origin={origin} />);
+
+      const canonicalLink = document.head.querySelector(
+        'link[rel="canonical"]'
+      );
+      expect(canonicalLink?.getAttribute('href')).toBe(`${origin}/en`);
+
+      const alternateFi = document.head.querySelector(
+        'link[rel="alternate"][hrefLang="fi"]'
+      );
+      expect(alternateFi?.getAttribute('href')).toBe(`${origin}/fi`);
+
+      const alternateXDefault = document.head.querySelector(
+        'link[rel="alternate"][hrefLang="x-default"]'
+      );
+      expect(alternateXDefault?.getAttribute('href')).toBe(`${origin}/fi`);
+    });
+
+    it('should handle path with trailing slash and query params', () => {
+      router.locale = 'sv';
+      router.asPath = '/test-path/?a=1';
+
+      render(<RouteMeta origin={origin} />);
+
+      const canonicalLink = document.head.querySelector(
+        'link[rel="canonical"]'
+      );
+      expect(canonicalLink?.getAttribute('href')).toBe(
+        `${origin}/sv/test-path`
+      );
+
+      const alternateSv = document.head.querySelector(
+        'link[rel="alternate"][hrefLang="sv"]'
+      );
+      expect(alternateSv?.getAttribute('href')).toBe(`${origin}/sv/test-path/`);
+    });
+  });
+
+  describe('with page prop', () => {
+    const mockPage = {
+      id: '1',
+      title: 'Test Page',
+      slug: 'test-page',
+      uri: '/fi/test-page/',
+      link: '/fi/test-page/',
+      contentTypeName: 'Page',
+      date: '2023-01-01',
+      modified: '2023-01-01',
+      language: {
+        code: 'FI',
+        id: '1',
+        locale: 'fi',
+        name: 'Suomi',
+        slug: 'fi',
+      },
+      translations: [
+        {
+          uri: '/sv/test-sida/',
+          language: { slug: 'sv' },
+        },
+        {
+          uri: '/en/test-page/',
+          language: { slug: 'en' },
+        },
+      ],
+    } as unknown as PageType;
```

Comment:
Removing unneeded fields (date, modified) and adding language.id fields, using enum for code makes the types for mockPage pass:
```suggestion
    const mockPage = {
      id: '1',
      title: 'Test Page',
      slug: 'test-page',
      uri: '/fi/test-page/',
      link: '/fi/test-page/',
      __typename: 'Page',
      language: {
        code: LanguageCodeEnum.Fi,
        id: '1',
        locale: 'fi',
        name: 'Suomi',
        slug: 'fi',
      },
      translations: [
        {
          uri: '/sv/test-sida/',
          language: { id: '1', slug: 'sv' },
        },
        {
          uri: '/en/test-page/',
          language: { id: '2', slug: 'en' },
        },
      ],
    } as const satisfies PageType;
```